src/naturallanguage.cs

```diff
@@ -0,0 +1,246 @@
+
+		[Export ("processString:")]
+		void Process (string @string);
+
+		[Export ("reset")]
+		void Reset ();
+
+		[Internal]
+		[NullAllowed, Export ("dominantLanguage")]
+		NSString _DominantLanguage { get; }
+
+		[Wrap ("NLLanguageExtensions.GetValue (_DominantLanguage)")]
+		NLLanguage DominantLanguage { get; }
+
+		// left in case the user does not want to get a c# dict
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("languageHypothesesWithMaximum:")]
+		NSDictionary<NSString, NSNumber> GetNativeLanguageHypotheses (nuint maxHypotheses);
+
+		// left in case the user does not want to get a c# dict
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("languageHints", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSNumber> NativeLanguageHints { get; set; }
+
+		[Internal]
+		[Export ("languageConstraints", ArgumentSemantic.Copy)]
+		NSString[] _LanguageConstraints { get; set; }
+
+		NLLanguage[] LanguageConstraints {
+			[Wrap ("Array.ConvertAll (_LanguageConstraints, e => NLLanguageExtensions.GetValue (e))")]
+			get;
+			[Wrap ("_LanguageConstraints = Array.ConvertAll (value, e => NLLanguageExtensions.GetConstant (e))")]
+			set;
+		}
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface NLModelConfiguration : NSCopying, NSSecureCoding {
+		[Export ("type")]
+		NLModelType Type { get; }
+
+		[Internal]
+		[NullAllowed, Export ("language")]
+		NSString _Language { get; }
+
+		NLLanguage Language { [Wrap ("(_Language.Handle != IntPtr.Zero)? NLLanguageExtensions.GetValue (_Language) : NLLanguage.Undetermined")] get; }
+
+		[Export ("revision")]
+		nuint Revision { get; }
+
+		[Static]
+		[Export ("supportedRevisionsForType:")]
+		NSIndexSet GetSupportedRevisions (NLModelType type);
+
+		[Static]
+		[Export ("currentRevisionForType:")]
+		nuint GetCurrentRevision (NLModelType type);
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface NLModel {
+		[Static]
+		[Export ("modelWithContentsOfURL:error:")]
+		[return: NullAllowed]
+		NLModel Create (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("configuration", ArgumentSemantic.Copy)]
+		NLModelConfiguration Configuration { get; }
+
+		[Export ("predictedLabelForString:")]
+		[return: NullAllowed]
+		string GetPredictedLabel (string @string);
+
+		[Export ("predictedLabelsForTokens:")]
+		string[] GetPredictedLabels (string[] tokens);
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface NLTokenizer
+	{
+		[Export ("initWithUnit:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NLTokenUnit unit);
+
+		[Export ("unit")]
+		NLTokenUnit Unit { get; }
+
+		[NullAllowed, Export ("string", ArgumentSemantic.Retain)]
+		string String { get; set; }
+
+		[Internal]
+		[Export ("setLanguage:")]
+		void _SetLanguage (string language);
```

Comment:
use `NSString` here, otherwise you'll cause extra conversions as `GetConstant` returns an `NSString` instance